app/src/main/java/com/asfoundation/wallet/ui/widget/holder/TransactionHolder.java

```diff
@@ -261,17 +264,25 @@ private void setTypeIconVisibilityBasedOnDescription(TransactionDetails details,
     }
   }
 
-  private String getScaledValue(String valueStr, long decimals) {
-    // Perform decimal conversion
+  private String getScaledValue(String valueStr, long decimals, String currencySymbol) {
+    WalletCurrency walletCurrency = mapToWalletCurrency(currencySymbol);
     BigDecimal value = new BigDecimal(valueStr);
     value = value.divide(new BigDecimal(Math.pow(10, decimals)));
-    int scale = 4; //SIGNIFICANT_FIGURES - value.precision() + value.scale();
-    return value.setScale(scale, RoundingMode.HALF_UP)
-        .stripTrailingZeros()
-        .toPlainString();
+    return formatter.formatCurrency(value.doubleValue(), walletCurrency);
   }
 
   @Override public void onClick(View view) {
     onTransactionClickListener.onTransactionClick(view, transaction);
   }
+
+  private WalletCurrency mapToWalletCurrency(String currency) {
```

Comment:
There are multiple ocurrences of this method across the app. can we make only one?